lib/foreman_maintain/package_manager/yum.rb

```diff
@@ -101,18 +105,19 @@ def setup_protector(enabled)
       File.open(protector_config_file, 'w') { |file| file.puts config }
     end
 
-    def yum_action(action, packages, with_status: false, assumeyes: false)
+    def yum_action(action, packages, with_status: false, assumeyes: false, valid_exit_statuses: [0])
       yum_options = []
       packages = [packages].flatten(1)
       yum_options << '-y' if assumeyes
+      yum_options << '--disableplugin=foreman-protector'
```

Comment:
Will this option be only for `check_update`?

Reply:
No, this option will for all the yum related task.
With "--disableplugin=foreman-protector" in place there is no need to unlock package while performing yum related task.
We will need to revisit all the commands/scenarios where we are doing yum related task and will need to remove the step `Procedures::Packages::UnlockVersions` For example - While installing package - [code](https://github.com/theforeman/foreman_maintain/blob/992b0a327d884c6af0641427a3ff394e3363ba16/definitions/scenarios/packages.rb#L64).
I think we should do above refactoring in separate PR.

Reply:
> I think we should do above refactoring in separate PR.

Could you please create one refactoring issue for tracking this change?

Reply:
I have opened the issue for refactoring -  [#30393](https://projects.theforeman.org/issues/30393)

Reply:
It works as expected. I am not liking the idea of hardcoding `--disableplugin=foreman-protector` in yum_options. I still prefer the method to unlock packages and lock it again; as this feels like hack. 
Maybe if we want to save the extra logic of disable and enable plugin which has written in setup_protector ? you can override yum_options in setup_protector itself ?
With this we should change the description of `check-update` command to,
~~~
check-update    Check for available package updates in unlocked session
~~~

Reply:
I am not sure how we can override the yum_options in setup_protector, because currently yum_option variable is scoped for `def yum_action`. 

Let me tell you why I think '--disableplugin=foreman-protector' is better option. 

a) I did not found any scenario where we do yum action without unlocking packages(disabling foreman-protector) plugin. 
b) In the current code, I think there are two scenarios where foreman-maintain can leave packages in the unlocked state.
	

1. If you see the implementation here - https://github.com/theforeman/foreman_maintain/blob/c8be149a1ecb92885f8a19f882503a4209412372/definitions/scenarios/packages.rb#L67 (or any other scenario). So we are unlocking the packages but there can be the times that foreman-maintain fails or user can force terminate the process (ctrl +c). In this case, foreman-maintain will leave packages in unlocked state. 
2. We are not explicitly locking the packages in any scenario, rather we are relying on foreman-installer, thinking that it will do the job of locking the packages but what if the installer fails in between? then the package will still stay in the unlocked state. 

Using the `--disableplugin=foreman-protector` will make sure that the package will not be in unlock state ever.

Reply:
Also, with this fix, we can get rid of this kind of issues - [#30414](https://projects.theforeman.org/issues/30414).